src/AppShed/Extensions/StorageBundle/Controller/ApiController.php

```diff
@@ -0,0 +1,444 @@
+            $em->persist($api);
+            $em->flush();
+            $this->get('session')->getFlashBag()->add(
+                'success',
+                'Your changes were saved'
+            );
+            return $this->redirect($this->generateUrl('api_list', $appParams));
+        }
+
+        $data = [
+            'api' => $api,
+            'form'   => $form->createView(),
+            'appParams' => $appParams
+        ];
+        return $data;
+    }
+
+    /**
+     * @Route("/api/{uuid}/delete", name="api_delete")
+     * @param Api $api
+     * @param $uuid
+     * @ParamConverter("api", class="AppShed\Extensions\StorageBundle\Entity\Api", options={"uuid"="uuid"})
+     * @Method({"GET", "POST"})
+     * @return array
+     */
+    public function deleteAction(Request $request, Api $api, $uuid)
+    {
+        $app = $this->getApp($request);
+        $appParams = $this->getExtensionAppParameters($request);
+        if ($api->getApp() != $app) {
+            throw new NotFoundHttpException("Api with uuid $uuid not found");
+        }
+
+        $em = $this->getDoctrine()->getManager();
+        $em->remove($api);
+        $em->flush();
+
+        return $this->redirect($this->generateUrl('api_list', $appParams));
+    }
+
+    /**
+     * @Route("/api/{uuid}", name="api_show")
+     * @param Api $api
+     * @ParamConverter("api", class="AppShed\Extensions\StorageBundle\Entity\Api", options={"uuid"="uuid"})
+     * @Method({"GET", "POST"})
+     * @return JsonResponse
+
+     */
+    public function showAction(Request $request, Api $api)
+    {
+        $result = [];
+        switch ($api->getAction()) {
+            case Api::ACTION_SELECT: {
+                $result = $this->selectData($api, $request);
+            } break;
+            case Api::ACTION_INSERT: {
+                $result = $this->insertData($api, $request);
+            } break;
+            case Api::ACTION_UPDATE: {
+                $result = $this->updateData($api, $request);
+            } break;
+            case Api::ACTION_DELETE: {
+                $result = $this->deleteData($api, $request);
+            } break;
+        }
+        return new JsonResponse($result, 200);
+    }
+
+    private function deleteData(Api $api, Request $request) {
+        $result = [];
+
+        $filters = $request->request->get('filters', '');
+        $additionalFilters = $this->getAdditionalFilters($filters);
+        //GET FILTERED DATA
+        $storeData = $this->getDoctrine()->getManager()->getRepository('AppShedExtensionsStorageBundle:Data')->getDataForApi($api, $additionalFilters);
+
+        if ($api->getLimit()) {
+            $storeData = $this->limitResults($storeData, $api->getLimit());
+        }
+
+        $executed = 0;
+        if (count($storeData)) {
+            $em = $this->getDoctrine()->getManager();
+            foreach ($storeData as $k => $value) {
+                $em->remove($storeData[$k]);
+            }
+            $executed = 1;
```

Comment:
Lets change this!

If all ok: lets return {updatedRows: 10}

Otherwise: throw an exception, in most cases 400, with some useful comment